Parser/tokenizer.c

```diff
@@ -371,8 +371,8 @@ update_fstring_expr(struct tok_state *tok, char cur)
 {
     assert(tok->cur != NULL);
 
-    Py_ssize_t size = strlen(tok->cur);
     tokenizer_mode *tok_mode = &(tok->tok_mode_stack[tok->tok_mode_stack_index]);
+    Py_ssize_t size = tok->cur - tok_mode->f_string_multi_line_start;
```

Comment:
Do we set the right operand in all cases even if is single quoted fstring?

Reply:
I think so. We already have a few sites that depend on `f_string_multi_line_start` directly (instead of checking if it is NULL and falling back to this). Maybe that needs to be done more explicitly though.